client/webserver/http.go

```diff
@@ -190,6 +195,96 @@ func (s *WebServer) handleOrders(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// handleOrders is the handler for the /orders/export page request.
+func (s *WebServer) handleExportOrders(w http.ResponseWriter, r *http.Request) {
+	filter := new(core.OrderFilter)
+	err := r.ParseForm()
+	if err != nil {
+		s.writeAPIError(w, "export error: %v", err)
+		return
+	}
+
+	filter.Hosts = r.Form["hosts"]
+	assets := r.Form["assets"]
+	filter.Assets = make([]uint32, len(assets))
+	for k, assetId := range assets {
+		filter.Assets[k] = uint32(parseStrToInt(assetId))
+	}
+	statuses := r.Form["statuses"]
+	filter.Statuses = make([]order.OrderStatus, len(statuses))
+	for k, statusId := range statuses {
+		filter.Statuses[k] = order.OrderStatus(parseStrToInt(statusId))
+	}
+
+	ords, err := s.core.Orders(filter)
+	if err != nil {
+		s.writeAPIError(w, "Orders error: %v", err)
+		return
+	}
+	csvWriter := csv.NewWriter(w)
+	err = csvWriter.Write([]string{
+		"Trade",
+		"Market",
+		"Type",
+		"Status",
+		"Rate",
+		"Filled",
+		"Settled",
+		"Time",
+	})
+	if err != nil {
+		s.writeAPIError(w, "export error: %v", err)
+		return
+	}
+	for _, ord := range ords {
+		rate := float64(ord.Rate) / 1e8
+		tradeStr := fmt.Sprintf("%.8f %s → %.8f %s",
+			(float64(ord.Qty)/1e8)*rate, ord.QuoteSymbol,
+			float64(ord.Qty)/1e8, ord.BaseSymbol)
+
+		typeStr := ord.Type.String() + " "
+		if ord.Sell {
+			typeStr += "sell"
+		} else {
+			typeStr += "buy"
+		}
+		filled := float64(ord.Filled) / float64(ord.Qty) * 100
+		var settled float64
+		for _, match := range ord.Matches {
+			if match.IsCancel {
+				continue
+			}
+			if (match.Side == order.Maker && match.Status >= order.MakerRedeemed) ||
+				(match.Side == order.Taker && match.Status >= order.MatchComplete) {
+				if ord.Type == order.MarketOrderType && !ord.Sell {
+					settled += float64(match.Qty) * float64(match.Rate) * 1e-8
+				} else {
+					settled += float64(match.Qty)
+				}
+			}
+		}
+		srvTime := encode.UnixTimeMilli(int64(ord.Stamp))
+		err = csvWriter.Write([]string{
+			tradeStr, // Trade
+			fmt.Sprintf("%s-%s @ %s", ord.BaseSymbol, ord.QuoteSymbol, ord.Host), // Market
+			typeStr,                       // Type
+			ord.Status.String(),           // Status
```

Comment:
An unrecognized order status value will panic in String, although that shouldn't happen unless there's DB corruption. Still, I think it's time to change `(OrderStatus) String()` in dex/order/status.go so it does not panic and instead returns "invalid".

Reply:
I change it to returns "invalid" but it is not matched the test. So I revert it to the original. May be need someone else change the test first